### ISSUE TITLE
Feature requests: toggle to select only the beginning and end lines of the brackets or only lines of their content

### ISSUE THREAD

**Issue by rogeriodec** [facelessuser/BracketHighlighter#593]
There is an example with an if / end in Lua:

![image](https://user-images.githubusercontent.com/11199752/131219972-663e99b5-9019-4233-a9cd-e0d0e589198f.png)

The inner instructions (print) have 2 tabs ident, I want to fix them to only 1 tab. So I press `ctrl+alt+super+s` (Select text between brackets):

![image](https://user-images.githubusercontent.com/11199752/131220068-61158bba-5648-4e2a-86c9-e5910738e86c.png)

In this case, it's selecting also the first line (if), but I need only lines 2 and 3 (print):
![image](https://user-images.githubusercontent.com/11199752/131220139-4ebb05b7-49a5-45ba-8b5e-fd7eecd08cd2.png)

In a reverse case, I would like to select only the start and end lines (if / end), for example, to get rid of them and leave only the inner lines (print):

![image](https://user-images.githubusercontent.com/11199752/131220211-8f221cac-eacd-4b4e-aaad-5fec32291199.png)

**Comment by facelessuser** [comment]
These types of things are usually implemented with [BracketHighlighter plugins](https://facelessuser.github.io/BracketHighlighter/customize/#bracket-plugin-api).

One of the reasons these types of things are done with BH plugins is to allow people to create whatever functionality they need without having to touch the core BH code, and without having to wait for me to specifically implement and/or accept such functionality officially into the project. They can immediately get one up and running by simply creating a BH plugin file in a subfolder under `User` and manage their own custom use cases.

My intent is to not necessarily support every possible selection and augmentation functionality a user may desire, but provide some basics and enable people to create their own custom ones if desired.  Also, by creating hooks for things like selections via BH plugins, the intent is to also encourage pull requests.

---

TLDR; In short, if the intent is for me to add this into the core, it is probably a maybe from me with a low priority unless there was a lot of interest for such functionality. With that said, I am more than happy to provide guidance if someone were interested to attempt to write a local custom plugin (I would maybe be open to a pull request as well). I would use the [select BH plugin](https://github.com/facelessuser/BracketHighlighter/blob/master/bh_modules/bracketselect.py) as a guide. There is a bit of complexity there for various selection behaviors, so maybe using it as a guide, but creating a separate line selection plugin is the best route. I fear without some refactoring in the selection plugin, yet another use case would just make things even more complicated there.

@gir-bot remove S: triage
@gir-bot add T: feature, P: maybe, C: plugins
